plugins/inputs/t128_tank/reader.go

```diff
@@ -171,23 +171,28 @@ func (r *Reader) getIndex(indexPath string, defaultIndex index) (index, error) {
 		r.log.Debugf("index file path not provided, starting with default index %s", defaultIndex.string())
 		return defaultIndex, nil
 	}
-	content, err := os.ReadFile(indexPath)
-	if errors.Is(err, os.ErrNotExist) {
+	content, err := os.Open(indexPath)
+	if os.IsNotExist(err) {
 		r.log.Debugf("index file %s does not exist, starting with default index %s", indexPath, defaultIndex.string())
 		return defaultIndex, nil
 
 	} else if err != nil {
 		return defaultIndex, fmt.Errorf("encountered error reading index file, starting with default index %s: %s", defaultIndex.string(), err)
 	}
-
-	r.log.Debugf("found '%s' in index file", content)
-
-	index, err := newIndex(string(content))
-	if err != nil {
-		return defaultIndex, fmt.Errorf("encountered error while parsing index file content, starting with default index %s: %s", defaultIndex.string(), err)
+	defer content.Close()
+	scanner := bufio.NewScanner(content)
```

Comment:
Scanner makes me nervous because it can get stuck on long lines. We shouldn't see any, but weird things can happen. Can you instead just use the content you had and `strings.Split("\n", content)[0]` to get the first line?